### ISSUE TITLE
Write an ADR to document DNS hosting architecture decisions

### ISSUE THREAD

**Issue by zandercymatics** [cisagov/manage.get.gov#2912]
### Issue description

Our DNS hosting architecture is being implemented and we would like to add an ADR to document what decisions we've made

### Acceptance criteria

- [ ] An ADR that discusses the major architecture decisions

**Comment by h-m-f-t** [comment]
I don't think we need an ADR to document why we're using Cloudflare, but an ADR on our DNS hosting architecture and design seems appropriate. Marking this as refinement.
